packages/ui/src/badge/badge.tsx

```diff
@@ -0,0 +1,101 @@
+/**
+ * WordPress dependencies
+ */
+import { forwardRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { Box } from '../box';
+import { type BoxProps } from '../box/types';
+import { type BadgeProps } from './types';
+
+/**
+ * Default render function that renders a span element with the given props.
+ *
+ * @param props The props to apply to the HTML element.
+ */
+const DEFAULT_RENDER = ( props: React.ComponentPropsWithoutRef< 'span' > ) => (
+	<span { ...props } />
+);
+
+/**
+ * Maps intent values to Box backgroundColor and color props.
+ * Uses strong emphasis styles (as emphasis prop has been removed).
+ * @param intent
+ */
+const getIntentStyles = (
+	intent: BadgeProps[ 'intent' ]
+): Pick<
+	BoxProps,
+	'backgroundColor' | 'color' | 'borderColor' | 'borderWidth'
+> => {
```

Comment:
We're no longer returning the border-related props. But we could probably generalize this to expressing that "it returns some properties that are valid for box" without necessarily listing them. [`Partial`](https://www.typescriptlang.org/docs/handbook/utility-types.html#partialtype) is good for this, though it could probably work with just `BoxProps` as well (since these properties are all optional to begin with).

```suggestion
): Partial< BoxProps > => {
```